src/main/java/org/jenkinsci/maven/plugins/hpi/hosting/PreferLTS.java

```diff
@@ -0,0 +1,42 @@
+package org.jenkinsci.maven.plugins.hpi.hosting;
+
+import org.apache.maven.project.MavenProject;
+import org.jenkinsci.maven.plugins.hpi.hosting.VerificationMessage.Severity;
+
+public class PreferLTS implements VerificationRule {
+	@Override
+	public VerificationMessage validate(MavenProject project) {
+		MavenProject jenkinsParent = findPluginParentPom(project);
+		if (jenkinsParent == null) {
+			return new VerificationMessage(
+					"Your plugin doesn't seem to have the Jenkins parent pom in its POM hierarchy. "
+							+ "Jenkins plugins generally inherit from the Jenkins parent POM.",
+					Severity.WARNING);
+		}
+		String jenkinsVersion = jenkinsParent.getVersion();
+
+		// In 2.x plugin-pom became a standalone project https://github.com/jenkinsci/plugin-pom
+		// In that case, the Jenkins version is specified in the property "jenkins.version".
+		if(jenkinsVersion.startsWith("2."))
+		{
+			jenkinsVersion = (String) project.getProperties().get("jenkins.version");
+		}
+		if (jenkinsVersion.split("\\.").length != 3) {
+			return new VerificationMessage("Your plugin does not seem to have a LTS Jenkins release. In general, "
+					+ "it's preferable to use an LTS version as parent version.",
```

Comment:
We generally do backports of API additions with `@Restricted`, so that should be rarely relevant. That said, `1.X` for `X in (565, 580, 609, 625, 642)` and `2.7` should be fine too IMO.

Note that we've been recommending [that plugin developers "rally around specific versions" for half a decade](https://wiki.jenkins-ci.org/pages/viewpage.action?pageId=58917579), so it would be weird to not even recommend not choosing any arbitrary weekly release.